### ISSUE TITLE
1.0.0

### ISSUE THREAD

**Issue by fvictorio** [prettier-solidity/prettier-plugin-solidity#567]
The idea of this issue is to have a "master issue" with all the things we need to do before we can release a stable, 1.0.0 release:

- [ ] Decide what 1.0.0 means, and how we want to use semver
- [ ] Document that decision
- [x] Decide what to do with [issue 474](https://github.com/prettier-solidity/prettier-plugin-solidity/issues/474). I think "doing nothing and then fix that for 1.1.0" would be a fine decision, but if someone thinks that should be fixed _before_ the first stable release, I could agree with that too,

Some extra notes below.

## What does 1.0.0 mean

As usual, instead of thinking about this, I believe it's better to just do whatever prettier core decided to do. [This issue](https://github.com/prettier/prettier/issues/2201) has a very useful discussion, and I mostly agree with what the authors say there. According to one of them:

> We initially used 0.0.x and shipped frenetically in the beginnings. Then 0.x.0 when it stopped being a toy project and we could see it being used by real people. We used 1.0 to signify that prettier was ready to be used at large scale. Then 1.x in order to show big milestones like typescript and css support. And 1.x.y for "hot fixes" after a release.

In addition, there are things at the API level that are clearly breaking changes, like removing an option.

Here's another interesting quote, answering the question "What is the semver change when previously formatted code is formatted differently?":

> Any. Depends on the case. The more impact, the more bump. Technically, you could say that any change whatsoever is a breaking change. But only doing major releases won’t help anyone. And it would mask actual API breaking changes.

I agree with this idea of using patch version bumps for small formatting changes, and minor version bumps for significant formatting changes. It's not very rigorous, but I don't think we can have a hard rule here.

One thing more: prettier recommends users to pin the version they are using, and I think we should do the same thing.

**Comment by fvictorio** [comment]
Btw, if you agree with those points about versioning, I can make a PR adding that to the docs.

**Comment by fvictorio** [comment]
Oh, and for the record, I think the `compiler` option should still be marked as experimental after the 1.0.0 release. That means we could remove it or change it significantly without having to make a major version bump.

**Comment by Janther** [comment]
I have some ideas for the `compiler` option. I don't believe it should not stay as it is right now.

**Comment by Janther** [comment]
For `v1.0.0` I expect the plugin to support ranges within a document. That way we will also support `format on paste` and not just `format on save`.

**Comment by fvictorio** [comment]
> For `v1.0.0` I expect the plugin to support ranges within a document. That way we will also support `format on paste` and not just `format on save`.

Added to the checklist. I'm curious though, why doesn't that "just work"? I would expect that prettier to handle it without us having to do anything.

**Comment by Janther** [comment]
Maybe it works, but we don't have proper tests for it.

**Comment by fvictorio** [comment]
I think the latest item is blocked by this: https://github.com/prettier/prettier/issues/7639

@Janther do you think this is still a must for 1.0.0 given that? Or can we move that to a separate issue to be worked on whenever prettier supports that?

**Comment by Janther** [comment]
so that's why it wasn't working in my tests 🤣. Sure let's wait for the future.
